extensions/amp-ad-network-adsense-impl/0.1/responsive-state.js

```diff
@@ -103,35 +101,14 @@ export class ResponsiveState {
     return new ResponsiveState(element, true);
   }
 
-  /**
-   * Upgrades the ad unit to full-width responsive if it does not fall back to container width.
-   * @param {!Element} element
-   * @param {string} adClientId
-   * @return {!Promise<?ResponsiveState>} a promise that resolves when any upgrade is complete.
-   */
-  static maybeUpgradeToResponsive(element, adClientId) {
-    // For Desktop Full-width Ad request, it should fall back to container width.
-    if (
-      element.hasAttribute('data-auto-format') &&
-      !ResponsiveState.isLayoutViewportNarrow_(element)
-    ) {
-      return ResponsiveState.convertToContainerWidth_(element);
-    }
-
-    return ResponsiveState.maybeUpgradeToFullWidthResponsive(
-      element,
-      adClientId
-    );
-  }
-
   /**
    * Upgrades the ad unit to responsive if there is an opt-in setting in localstorage.
    * See https://github.com/ampproject/amphtml/issues/23568 for design.
    * @param {!Element} element
    * @param {string} adClientId
    * @return {!Promise<?ResponsiveState>} a promise that resolves when any upgrade is complete.
    */
-  static maybeUpgradeToFullWidthResponsive(element, adClientId) {
+  static maybeUpgradeToResponsive(element, adClientId) {
```

Comment:
is L121 redundant now?

Reply:
Don't think it's redundant.

The check in one level higher is for ContainerWidthResponsive. There are two conditions need to meet for ContainerWidthResponsive:
1. The user is on desktop.
2. The user is using full-width responsive tag.

Without the check in L121, for 1. The user is on desktop & 2. The user does `not` use full-width responsive tag, it will try to request Ad Size upgrade for the desktop users. As a result, it will create the broken user experience like the earlier reported bug.

Reply:
I see, thanks for clarifying!